copy/multiple.go

```diff
@@ -12,11 +12,46 @@ import (
 	internalManifest "github.com/containers/image/v5/internal/manifest"
 	"github.com/containers/image/v5/manifest"
 	"github.com/containers/image/v5/signature"
+	digest "github.com/opencontainers/go-digest"
 	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/exp/slices"
 )
 
+type CopyOperation int
+
+const (
+	Ignore CopyOperation = iota + 1
+	CopyWithoutModification
+	//TODO: following states must be implemented
+	//CopyWithModification
+	//CopyAndReplicate
+)
+
+type instanceCopy struct {
+	digest digest.Digest
+	op     CopyOperation
+}
+
+// instancesToCopy returns the list of digests which actually need to be copied to the new manifest list, ignoring the ones which are not needed.
+func (c *copier) instancesToCopy(options *Options, list internalManifest.List) (instances []instanceCopy, copyCount int) {
+	result := []instanceCopy{}
+	instanceDigests := list.Instances()
+	imagesToCopy := len(instanceDigests)
+	if options.ImageListSelection == CopySpecificImages {
+		imagesToCopy = len(options.Instances)
+	}
```

Comment:
(I wouldn’t mind counting the items one by one as the relevant operations are chosen, or possibly even doing one more iteration over the loop just to count.; I think we’ll end up doing one of these a future version anyway, and it’s more obviously correct without having to think — so we might as well start now.)

Reply:
Not sure if this is relevent anymore `EditInstances` directly take list of updates which can be directly updated.